problem_builder/models.py

```diff
@@ -72,3 +80,17 @@ class Meta(object):
         # specify the app_label here.
         app_label = 'problem_builder'
         unique_together = (('shared_by', 'shared_with', 'block_id'),)
+
+
+# Signals ###########################################################
+
+def delete_anonymous_user_answers(sender, **kwargs):
+    """
+    Delete Answer records when an AnonymousUserId is deleted.
+    """
+    instance = kwargs['instance']
+    Answer.objects.filter(student_id=instance.anonymous_user_id).delete()
+
+
+if ANONYMOUS_USER_ID_IMPORTED:
+    pre_delete.connect(delete_anonymous_user_answers, sender=AnonymousUserId)
```

Comment:
Here is an alternate approach which will make this easier to test:

Create a function called ``get_anonymous_user_id_model`` that imports and returns the AnonymousUserId model. This can be then stubbed by the tests to use a different model or a mock to test it. 

As it stands these tests will only run in edx-platform I think.

Reply:
@xitij2000 Hmm I've tried mocking the model, but it didn't work as expected since the condition to `connect` the signal happens when the `models.py` module is loaded, so the mock didn't have the chance to replace it yet.

Here's what I've tried:

**`models.py`**
```python
def get_anonymous_user_id_model():
    """
    Get AnonymousUserId via a function so it can be mocked in tests.
    """
    try:
        from student.models import AnonymousUserId
        return AnonymousUserId
    except ImportError:
        pass
...
if get_anonymous_user_id_model():
    pre_delete.connect(delete_anonymous_user_answers, sender=get_anonymous_user_id_model())
```

Maybe I'm missing something?

Also, the `delete_anonymous_user_answers` function itself is already tested by [this test](https://github.com/open-craft/problem-builder/pull/225/files#diff-87000eee015fac540c3d5af65ad462c5R36). I'm wondering if we still need to test the condition that `connect`s the signal receiver?

Reply:
@john2x Yeah, you're right, the main thing is tested anyway, and signal handling is a core mechanic so doesn't really need testing. 

I'll test is manually in the devstack, I think that should be enough.